packages/ragbits-document-search/src/ragbits/document_search/documents/element.py

```diff
@@ -90,23 +90,35 @@ def from_vector_db_entry(cls, db_entry: VectorStoreEntry) -> "Element":
         """
         element_type = db_entry.metadata["element_type"]
         element_cls = Element._elements_registry[element_type]
+        if "embedding_type" in db_entry.metadata:
+            del db_entry.metadata["embedding_type"]
         return element_cls(**db_entry.metadata)
 
-    def to_vector_db_entry(self, vector: list[float]) -> VectorStoreEntry:
+    def to_vector_db_entry(self, vector: list[float], embedding_type: EmbeddingTypes | None = None) -> VectorStoreEntry:
         """
         Create a vector database entry from the element.
 
         Args:
             vector: The vector.
+            embedding_type: EmbeddingTypes.TEXT, EmbeddingTypes.IMAGE or None
 
         Returns:
             The vector database entry
         """
+        metadata = self.model_dump(exclude={"id", "key"})
+        vector_store_entry_id = self.id
+        if embedding_type:
+            id_components = [
+                vector_store_entry_id,
+                str(embedding_type),
+            ]
+            vector_store_entry_id = str(uuid.uuid5(uuid.NAMESPACE_OID, ";".join(id_components)))
```

Comment:
if we calculate the actual id here, we don't need the `id` attribute at class level

Reply:
it is somehow different - here we compute id for vector store entry: eg. we have an image element that has description and ocr and image supporting embedding - those we want to create 2 entries in the store for embedding: one for descriptions, second for image and we want to distinguish between them - while Element itself remains the same

Reply:
I get it, but this calculation is redundant since we use this id to distinguish vector store entries, not processor elements

Reply:
I'd even put this `id` computation into `VectorStoreEntry` class, but the implementation might be a bit shady since we'd need to rely on dynamic metadata field

Reply:
but this computation does nothing with processor element - it does not modify it, so i've done it here as kind of workaroung of computing it while instantiation of `VectorStoreEntry` (we could ofc. have separate field embedding_type, however imo it is metadata field)

Reply:
I'm just saying that calculating uuid5 twice for a vector entry id that is (currently) stored in an element is ambiguous, and everything should be calculated here or move it to the element class.

Reply:
done with instantiation of `VectorStoreEntry` with class method. could have been `from_element` as well i guess, but than we would have some issues with nested imports.

Reply:
solved in a way discussed on a meeting